### PR TITLE
feat: [IAC-1228]: Add the ability to show IACM policy evaluations

### PR DESCRIPTION
**Pull request by willd0yle**
### Summary
Add iacm to policy evaluations screen

#### Screenshots
Screenshot 2023-11-09 at 12 57 41

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

File: src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/ExecutionPolicyEvaluationsView.tsx [hunk 1]
```diff
@@ -5,25 +5,52 @@
...
+import { useListIACMEvaluationsForExecution } from './hooks/useIacmGovernance'
+import { EvaluationStatus } from './types'
+import { formatIacmGovernanceData } from './utils'
 
 export default function ExecutionPolicyEvaluationsView(): React.ReactElement | null {
+  const IACM_ENABLED = useFeatureFlag(FeatureFlag.IACM_ENABLED)
   const { accountId, module } = useParams<PipelineType<ExecutionPathProps>>()
   const context = useExecutionContext()
   const governanceMetadata = context?.pipelineExecutionDetail?.pipelineExecutionSummary?.governanceMetadata
+  const { data } = useListIACMEvaluationsForExecution()
```
Comment (bgribben-harness) [inline src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/ExecutionPolicyEvaluationsView.tsx:26]: Do we need to handle loading/error states?

File: src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/ExecutionPolicyEvaluationsView.tsx [hunk 2]
```diff
@@ -5,25 +5,52 @@
...
 
-  if (!governanceMetadata) {
-    return null
-  }
```
Comment (bgribben-harness) [inline src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/ExecutionPolicyEvaluationsView.tsx:22]: What happens now if there is no `governanceMetadata`? Can the `PolicyManagementEvaluationView` handle this?

File: src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/ExecutionPolicyEvaluationsView.tsx [hunk 3]
```diff
@@ -5,25 +5,52 @@
...
+  const metaData = useMemo(() => {
+    if (!data && !governanceMetadata) return null
+
+    const details: any = []
+    let status: EvaluationStatus = EvaluationStatus.PASS
+    if (governanceMetadata) {
+      const { details: opaDetails, status: opaStatus } = governanceMetadata
+      map(opaDetails, policySet => details.push(policySet))
```
Comment (bgribben-harness) [inline src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/ExecutionPolicyEvaluationsView.tsx:35]: You could use `opaDetails.forEach` here if you're not building a new array

Comment (bgribben-harness) [reply src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/ExecutionPolicyEvaluationsView.tsx:35]: Also could this just be 
```js
let details: any = [];
...
details = [...opaDetails] 
```

File: src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/ExecutionPolicyEvaluationsView.tsx [hunk 4]
```diff
@@ -5,25 +5,52 @@
...
+      if (status !== opaStatus) {
+        status = includes(opaStatus, EvaluationStatus.ERROR) ? EvaluationStatus.ERROR : EvaluationStatus.WARNING
```
Comment (bgribben-harness) [inline src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/ExecutionPolicyEvaluationsView.tsx:37]: This seems odd; is `opaStatus` an array? If not, why not just do a string comparison?

File: src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/ExecutionPolicyEvaluationsView.tsx [hunk 5]
```diff
@@ -5,25 +5,52 @@
...
+      }
+    }
+
+    if (IACM_ENABLED && data) {
+      const { iacmData, iacmStatus } = formatIacmGovernanceData(data)
+      map(iacmData, policySet => details.push(policySet))
```
Comment (bgribben-harness) [inline src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/ExecutionPolicyEvaluationsView.tsx:43]: As above

File: src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/ExecutionPolicyEvaluationsView.tsx [hunk 6]
```diff
@@ -5,25 +5,52 @@
...
+      if (status !== iacmStatus) {
+        status = includes(iacmStatus, EvaluationStatus.ERROR) ? EvaluationStatus.ERROR : EvaluationStatus.WARNING
```
Comment (bgribben-harness) [inline src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/ExecutionPolicyEvaluationsView.tsx:45]: If this status variable represents the overall status of the evaluations, you could calculate once at the end of this block with a `.some`

File: src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/hooks/useIacmGovernance.tsx [hunk 7]
```diff
@@ -0,0 +1,53 @@
+import { UseQueryResult, useQuery } from '@tanstack/react-query'
+import { useParams } from 'react-router-dom'
+import { ProjectPathProps } from '@modules/10-common/interfaces/RouteInterfaces'
+import { Evaluation } from 'services/pm'
+import { resolveUrl } from 'services/sto/fetcher'
+import SessionToken from 'framework/utils/SessionToken'
+
+export const queryFn = (url: string) => async () => {
```
Comment (bgribben-harness) [inline src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/hooks/useIacmGovernance.tsx:8]: Probably best not to export this if it's only used within this hook

File: src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/utils.ts [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+      })
+    })
+  )
+  return {
+    iacmData,
+    iacmStatus: includes(statuses, EvaluationStatus.ERROR)
+      ? EvaluationStatus.ERROR
+      : includes(statuses, EvaluationStatus.WARNING)
+      ? EvaluationStatus.WARNING
+      : EvaluationStatus.PASS
```
Comment (bgribben-harness) [inline src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/utils.ts:44]: Nested ternaries are evil

File: src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/ExecutionPolicyEvaluationsView.tsx (hunk 1 above)
Comment (willd0yle) [reply src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/ExecutionPolicyEvaluationsView.tsx:26]: Still needs to be added.

File: src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/ExecutionPolicyEvaluationsView.tsx (hunk 2 above)
Comment (willd0yle) [reply src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/ExecutionPolicyEvaluationsView.tsx:22]: Need to test the potential error states and empty states.

File: src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/ExecutionPolicyEvaluationsView.tsx (hunk 4 above)
Comment (willd0yle) [reply src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/ExecutionPolicyEvaluationsView.tsx:37]: Used to be an array which renders this useless and will update.

File: src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/__tests__/ExecutionPolicyEvaluationsView.test.tsx [hunk 9]
```diff
@@ -51,4 +61,126 @@ describe('<ExecutionPolicyEvaluationsView /> tests', () => {
...
+
+    const { container } = render(
+      <TestWrapper defaultFeatureFlagValues={{ IACM_ENABLED: true }}>
+        <ExecutionPolicyEvaluationsView />
+      </TestWrapper>
+    )
+    expect(container).toMatchSnapshot()
+  })
+
+  test('Should render empty when iacm data is not null', () => {
```
Comment (bgribben-harness) [inline src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/__tests__/ExecutionPolicyEvaluationsView.test.tsx:126]: Is this supposed to be the same as the previous test?

**Review by bgribben-harness — APPROVED** [review]

File: src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/__tests__/ExecutionPolicyEvaluationsView.test.tsx (hunk 9 above)
Comment (willd0yle) [reply src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/__tests__/ExecutionPolicyEvaluationsView.test.tsx]: updated the test

**Review by bgribben-harness — APPROVED** [review]

**Review by bgribben-harness — DISMISSED** [review]

File: src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/types.ts
Comment (mayankVermaHarness) [inline src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/types.ts:1]: Are these "types" not exported by a service file?

File: src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/types.ts [hunk 10]
```diff
@@ -0,0 +1,32 @@
+export enum EvaluationStatus {
+  ERROR = 'error',
+  PASS = 'pass',
+  WARNING = 'warning'
+}
+
+export interface IacmFormatedData {
```
Comment (mayankVermaHarness) [inline src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/types.ts:7]: Nit: 
```suggestion
export interface IacmFormattedData {
```

**Review by mayankVermaHarness — APPROVED** [review]

File: src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/types.ts
Comment (willd0yle) [reply src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/types.ts:1]: The governance meta data type which is what this data is being modelled after is actually `[key: string]: any` so I made the types to make it easier to use.

**Comment by willd0yle** [comment]
retrigger jest

**Review by mayankVermaHarness — APPROVED** [review]

File: src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/ExecutionPolicyEvaluationsView.tsx [hunk 11]
```diff
@@ -5,25 +5,55 @@
...
+    includeChildScopes: true,
+    module: 'iacm',
+    entity: `iacm-${executionIdentifier}`,
+    lazy: !IACM_ENABLED
+  }
+  const { data, loading, error } = useGetEvaluationList({ queryParams })
+
+  useEffect(() => {
+    if (error) {
+      showError(error)
```
Comment (abhinavrastogi-harness) [inline src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/ExecutionPolicyEvaluationsView.tsx:39]: this can print unhelpful errors as well to the toaster. 
we should wrap this in util functions `getErrorInfoFromErrorObject` and `shouldShowError` from UICore

File: src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/__tests__/ExecutionPolicyEvaluationsView.test.tsx [hunk 12]
```diff
@@ -51,4 +61,106 @@ describe('<ExecutionPolicyEvaluationsView /> tests', () => {
...
+    mockImport('@governance/PolicyManagementEvaluationView', {
+      PolicyManagementEvaluationView: () => <div /> // eslint-disable-line react/display-name
+    })
+
+    const { container } = render(
+      <TestWrapper defaultFeatureFlagValues={{ IACM_ENABLED: true }}>
+        <ExecutionPolicyEvaluationsView />
+      </TestWrapper>
+    )
+    expect(container).toMatchSnapshot()
```
Comment (abhinavrastogi-harness) [inline src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/__tests__/ExecutionPolicyEvaluationsView.test.tsx:121]: please use assertions instead of snapshots

File: src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/__tests__/ExecutionPolicyEvaluationsView.test.tsx [hunk 13]
```diff
@@ -51,4 +61,106 @@ describe('<ExecutionPolicyEvaluationsView /> tests', () => {
...
```
Comment (abhinavrastogi-harness) [inline src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/__tests__/ExecutionPolicyEvaluationsView.test.tsx:164]: please use assertions instead of snapshots

File: src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/__tests__/__snapshots__/ExecutionPolicyEvaluationsView.test.tsx.snap [hunk 14]
```diff
@@ -12,3 +12,25 @@ exports[`<ExecutionPolicyEvaluationsView /> tests Should render empty when gover
...
 
 exports[`<ExecutionPolicyEvaluationsView /> tests Should render empty when governanceMetadata is null 1`] = `<div />`;
+
+exports[`<ExecutionPolicyEvaluationsView /> tests Should render empty when iacm data status of error 1`] = `
+<div>
+  <div
+    class="StyledProps--font StyledProps--main"
+    style="width: 100%; height: 100%;"
+  >
+    <div />
```
Comment (abhinavrastogi-harness) [inline src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/__tests__/__snapshots__/ExecutionPolicyEvaluationsView.test.tsx.snap:22]: what's the point of a test which renders empty?

**Review by abhinavrastogi-harness — APPROVED** [review]

File: src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/__tests__/ExecutionPolicyEvaluationsView.test.tsx (hunk 13 above)
Comment (udayvunnam) [reply src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/__tests__/ExecutionPolicyEvaluationsView.test.tsx:164]: +100

**Comment by udayvunnam** [comment]
Tests need some cleanup/improvement, otherwise LGTM

**Review by udayvunnam — APPROVED** [review]
